lib/sensors/src/temperature.rs

```diff
@@ -32,20 +40,25 @@ impl<'a, T: HypedI2c> Temperature<'a, T> {
     /// Read the temperature from the sensor and return it as a floating point value in degrees Celsius
     pub fn read(&mut self) -> Option<f32> {
         // Read the high and low bytes of the temperature and combine them to get the temperature
-        let temperature_high_byte =
-            match self.i2c.read_byte(self.device_address, STTS22H_DATA_TEMP_H) {
-                Some(byte) => byte,
-                None => {
-                    return None;
-                }
-            };
-        let temperature_low_byte =
-            match self.i2c.read_byte(self.device_address, STTS22H_DATA_TEMP_L) {
-                Some(byte) => byte,
-                None => {
-                    return None;
-                }
-            };
+        let temperature_high_byte = match self.i2c.lock(|i2c| {
+            i2c.borrow_mut()
+                .read_byte(self.device_address, STTS22H_DATA_TEMP_H)
+        }) {
+            Some(byte) => byte,
+            None => {
+                return None;
+            }
+        };
+
+        let temperature_low_byte = match self.i2c.lock(|i2c: &RefCell<T>| {
+            i2c.borrow_mut()
+                .read_byte(self.device_address, STTS22H_DATA_TEMP_L)
+        }) {
```

Comment:
is it possible to move the `Mutex<Refcell<...>>` into the `i2c` impl so this is done there instead of needing done on every call in every sensor? maybe wrapping the `i2c` field of `Stm32f767ziI2c` in the `Mutex<RefCell<...>>` or something similar?

Reply:
Good shout, I've changed this. Still a bit clunky because we still have to pass in a mutable reference to the sensors to allow us to read `MockI2c` values afterwards